compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -82,14 +82,17 @@ impl QueryContext for QueryCtxt<'_> {
         tls::with_related_context(self.tcx, |icx| icx.query)
     }
 
-    fn collect_active_jobs(self) -> QueryMap {
+    /// Returns a query map representing active query jobs and a bool being false
+    /// if there was an error constructing the map.
+    fn collect_active_jobs(self) -> (QueryMap, bool) {
```

Comment:
```suggestion
    fn collect_active_jobs(self) -> Result<QueryMap, QueryMap> {
```

Might be more explicit. When not caring about whether the map is complete or not one can do:

```rs
let query_map = match qcx.collect_active_jobs() {
  Ok(complete_map) => complete_map,
  Err(incomplete_map) => incomplete_map,
};
```

Reply:
That does feel a bit odd to me since `QueryMap` is not an error.

Reply:
Using `QueryMap` as an error does force callers of `collect_active_jobs` to check if it is incomplete or not, the `bool` can still easily be ignored by accident. Pretty sure i've seen this pattern of `Result<T, T>` in rustc and elsewhere.

Reply:
If we use `Result<QueryMap, QueryMap>`, we should at least do something about the incomplete `QueryMap`, but we doesn't. I've not had to deal with an incomplete `QueryMap` in practice.